spring-shell2-shell1-adapter/src/main/java/org/springframework/shell2/legacy/LegacyParameterResolver.java

```diff
@@ -113,17 +112,67 @@ public ParameterDescription describe(MethodParameter parameter) {
 
 	@Override
 	public List<CompletionProposal> complete(MethodParameter parameter, CompletionContext context) {
-		return null;
+		// DevNote: in Legacy the user is either completing an argument (-- prefix) or the value of the argument (e.g. no arity to consider).
+		// Other cases to be considered include default keys and default values
+		
+		CliOption option = parameter.getParameterAnnotation(CliOption.class);
+		List<String> keys = Arrays.asList(option.key());
+		
+		int currentWordIndex = context.getWordIndex();		
+		
+		// 1) has a key for this parameter already been defined?
+		boolean keyAlreadyDefined = context.getWords().stream()
```

Comment:
> Only ParameterResolvers know how a set parameter looks like

Agree on this point. 

To add to this, I'm going to add the items below to check if I'm understanding this in the correct way, maybe it will help pin-point why I might be coming from a different angle :) :

1. The arguments to invoke a single method could be resolved by different `ParameterResolver`s (currently `JLineShell::resolveArgs` allows this, so I'm using this as an assumption).
1. `ParameterResolver`s are expected to be state-less by `JLineShell`. As in, a single instance of a given `ParameterResolver` is used to resolve multiple parameters inside the `JLineShell::resolveArgs` method. 
1. An invocation to `ParameterResolver::complete(param0, ...)` shouldn't need to invoke `ParameterResolver::complete(paramX, ...)` or `ParameterResolver::resolve(paramX, ...)` (where X != 0 :) ). It might rely or make use of pre-computed results, but shouldn't need to invoke them directly (potential big question mark on this one?). Main reason I say this is due to item 1 above, the `ParameterResolver` for `param0` could be different to the one being used for `param1`.
1. Currently `ParameterResolver`s don't share their results during a completion/resolve request (for lack of a better term). This makes it a bit more tricky as `JLineShell` invokes the `ParameterResolver`s in order of definition in the method, but the user could've provided them (or be providing them) in a different order (this can also make #48 a bit more complex?).

If any of the above aren't right then it might help spot where the difference is coming from :)

Using the example again of the `register module --force <TAB>` that I use below, `--force` is the last parameter in the method, and other parameters don't know if it has already been completed or not. So, when JLineShell requests completions for the `type` parameter, it doesn't know if it should provide the keys as completions or not. If `--force` has a `specifiedDefaultValue` defined, then it should, but should the completion for the `type` parameter be checking those details of the `--force` parameter? Specially when considering item 1 above, it might not know how to.

> I think this is not as easy as it sounds

Agree on this as well :) hence why I want to double check we're on the same page. These look like core concepts I need to get the hang of. Thanks for the review and all the input!